pkg/invariants/panic.go

```diff
@@ -0,0 +1,78 @@
+package invariants
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"k8s.io/client-go/rest"
+
+	"github.com/openshift/origin/pkg/monitor/monitorapi"
+	"github.com/openshift/origin/pkg/test/ginkgo/junitapi"
+)
+
+func startCollectionWithPanicProtection(ctx context.Context, invariantTest InvariantTest, adminRESTConfig *rest.Config, recorder monitorapi.RecorderWriter) (err error) {
+	defer func() {
+		if r := recover(); r != nil {
+			err = fmt.Errorf("caught panic: %v", r)
```

Comment:
Does this mean if these panic we potentially don't end up with any junits for it?